src/crafting_gui.cpp

```diff
@@ -48,6 +51,40 @@ void load_recipe_category( JsonObject &jsobj )
     }
 }
 
+std::string get_cat_name( std::string prefixed_name )
+{
+    if( prefixed_name.find("CC_") == 0 ) {
+        return prefixed_name.substr( 3, prefixed_name.size() - 3 );
+    }
+
+    debugmsg( ( "could not get screen name of: " + prefixed_name ).c_str() );
+    return prefixed_name;
+}
+
+std::string get_subcat_name( const std::string &cat, std::string prefixed_name )
+{
+    std::string prefix = "CSC_" + get_cat_name( cat ) + "_";
+
+    if( prefixed_name.find( prefix ) == 0 ) {
+        return prefixed_name.substr( prefix.size(), prefixed_name.size() - prefix.size() );
+    } else if( prefixed_name.find("CSC_ALL") == 0 ) {
+        return "ALL";
+    }
+
+    debugmsg( ( "could not get screen name of: " + prefixed_name ).c_str() );
```

Comment:
Good idea, I was a little scared of messing with the load function, since I didn't make it.